_integration-tests/tests/grpc/grpc.go

```diff
@@ -31,17 +32,50 @@ func (tc *TestCase) Setup(t *testing.T) {
 	require.NoError(t, err)
 	tc.addr = lis.Addr().String()
 
-	tc.Server = grpc.NewServer()
+	var (
+		interceptedDirect atomic.Bool
+		interceptedChain  atomic.Bool
+	)
+	tc.Server = grpc.NewServer(
+		// Register a bunch of interceptors to ensure ours does not cause a runtime crash.
+		grpc.UnaryInterceptor(func(ctx context.Context, req any, _ *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp any, err error) {
+			interceptedDirect.Store(true)
+			return handler(ctx, req)
+		}),
+		grpc.ChainUnaryInterceptor(func(ctx context.Context, req any, _ *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (resp any, err error) {
+			interceptedChain.Store(true)
+			return handler(ctx, req)
+		}),
+	)
 	helloworld.RegisterGreeterServer(tc.Server, &server{})
 
 	go func() { assert.NoError(t, tc.Server.Serve(lis)) }()
 	t.Cleanup(func() {
 		tc.Server.GracefulStop()
+		assert.True(t, interceptedDirect.Load(), "original interceptor was not called")
+		assert.True(t, interceptedChain.Load(), "original chained interceptor was not called")
```

Comment:
@RomainMuller Are the messages correct? If `intercepted*` are `true`, doesn't it mean they were called? The same for the same messages below.

Reply:
The message is for when it fails... But I'm a litle ambivalent as to whether they should mention what should've happened or what was observed that is a failure

Reply:
Oh, you are right! I usually go with something like `original interceptor expected a call, it didn't happen`. Anyway, it's fine as long as it's understandable in the testing logs' context.